lib/easypost/beta/referral.rb

```diff
@@ -2,18 +2,71 @@
 
 # Referral objects are User objects created from a Partner user.
 class EasyPost::Beta::Referral < EasyPost::Resource
+  class << self
+    protected
+
+    # Retrieve EasyPost's Stripe public API key.
+    def retrieve_easypost_stripe_api_key
+      response = EasyPost.make_request(:get, '/beta/partners/stripe_public_key', @api_key)
+      response['public_key']
+    end
+
+    # Get credit card token from Stripe.
+    def create_stripe_token(number, expiration_month, expiration_year,
+                            cvc, easypost_stripe_token)
+      headers = {
+        # This Stripe endpoint only accepts URL form encoded bodies.
+        Authorization: "Bearer #{easypost_stripe_token}",
+        'Content-type': 'application/x-www-form-urlencoded',
+      }
+
+      credit_card_hash = {
+        card: {
+          number: number,
+          exp_month: expiration_month,
+          exp_year: expiration_year,
+          cvc: cvc,
+        },
+      }
+
+      form_encoded_params = EasyPost::Util.form_encode_params(credit_card_hash)
+
+      uri = URI.parse('https://api.stripe.com/v1/tokens')
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = true
+      request = Net::HTTP::Post.new(uri.request_uri, headers)
+      query = URI.encode_www_form(form_encoded_params)
+
+      response = http.request(request, query)
+      response_json = JSON.parse(response.body)
+      response_json['id']
+    end
+
+    # Submit Stripe credit card token to EasyPost.
+    def create_easypost_credit_card(referral_api_key, stripe_object_id, priority = 'primary')
+      wrapped_params = {
+        credit_card: {
+          stripe_object_id: stripe_object_id,
+          priority: priority,
+        },
+      }
+      response = EasyPost.make_request(:post, '/beta/credit_cards', referral_api_key, wrapped_params)
+      EasyPost::Util.convert_to_easypost_object(response, referral_api_key)
+    end
+  end
+
   # Create a referral user. This function requires the Partner User's API key.
   # <b>DEPRECATED:</b> Please use <tt>Referral</tt> in the main namespace instead.
   def self.create(params = {}, api_key = nil)
-    warn '[DEPRECATION] Please use `EndShipper.create` in the main namespace instead.'
+    warn '[DEPRECATION] Please use `Referral.create` in the main namespace instead.'
     response = EasyPost.make_request(:post, '/beta/referral_customers', api_key, { user: params })
```

Comment:
Question: Since the functionality of beta vs. GA EndShipper didn't change, can't we just re-route these deprecated beta functions to use the GA functions, rather than having to re-implement them?

Reply:
Most likely, will investigate

Reply:
Oh well unlike an incorrectly named function like we've done before, the behavior varies slightly between beta and GA so we need to keep the beta implementation here for that reason (also they need to be hitting the beta endpoints), otherwise we lose the whole reason we split up the namespaces to "pin" the behavior while it's available.

TLDR: We can't simply point the one to the other in this case.